src/Test.sol

```diff
@@ -139,6 +139,256 @@ abstract contract Test is DSTest {
             addr := create(0, add(bytecode, 0x20), mload(bytecode))
         }
     }
+
+    /*//////////////////////////////////////////////////////////////////////////
+                                    APPROX EQUAL
+    //////////////////////////////////////////////////////////////////////////*/
+
+    function assertApproxEq(
+        uint256 a,
+        uint256 b,
+        uint256 maxDelta
+    ) internal virtual {
+        uint256 delta = a > b ? a - b : b - a;
+
+        if (delta > maxDelta) {
+            emit log            ("Error: a ~= b not satisfied [uint]");
+            emit log_named_uint ("  Expected", b);
+            emit log_named_uint ("    Actual", a);
+            emit log_named_uint (" Max Delta", maxDelta);
+            emit log_named_uint ("     Delta", delta);
+            fail();
+        }
+    }
+
+    function assertApproxEq(
+        uint256 a,
+        uint256 b,
+        uint256 maxDelta,
+        string memory err
+    ) internal virtual {
+        uint256 delta = a > b ? a - b : b - a;
+
+        if (delta > maxDelta) {
+            emit log_named_string   ("Error", err);
+            emit log_named_uint     ("  Expected", b);
+            emit log_named_uint     ("    Actual", a);
+            emit log_named_uint     (" Max Delta", maxDelta);
+            emit log_named_uint     ("     Delta", delta);
+            fail();
+        }
+    }
+    function assertApproxEq(
+        int256 a,
+        int256 b,
+        int256 maxDelta
+    ) internal virtual {
+        int256 delta = a > b ? a - b : b - a;
+
+        if (delta > maxDelta) {
+            emit log            ("Error: a ~= b not satisfied [int]");
+            emit log_named_int  ("  Expected", b);
+            emit log_named_int  ("    Actual", a);
+            emit log_named_int  (" Max Delta", maxDelta);
+            emit log_named_int  ("     Delta", delta);
+            fail();
+        }
+    }
+
+    function assertApproxEq(
+        int256 a,
+        int256 b,
+        int256 maxDelta,
+        string memory err
+    ) internal virtual {
+        int256 delta = a > b ? a - b : b - a;
+
+        if (delta > maxDelta) {
+            emit log_named_string   ("Error", err);
+            emit log_named_int      ("  Expected", b);
+            emit log_named_int      ("    Actual", a);
+            emit log_named_int      (" Max Delta", maxDelta);
+            emit log_named_int      ("     Delta", delta);
+            fail();
+        }
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////
+                                    REL APPROX EQUAL
+    //////////////////////////////////////////////////////////////////////////*/
```

Comment:
Nit, but really just need one section header for all the assertions to differentiate them from the other helper methods. Maybe just called "Assertions"

Reply:
Due to the scope for a huge amount of overloading, I found code regions helpful for ensuring proper organisation of related but overloaded functions. That said, if people (read maintainers of forge-std) prefer to drop down to just one ASSERTIONS heading then happy for that change to get rolled in prior to this merging

Reply:
> only hesitation would be if we wanted to properly overload `assertApproxEqRel` and `assertApproxEqAbs`

By "properly overload" do you mean with types other than int256/uint256? If so, I think just supporting int256/uint256 would cover most cases and is sufficient.

Either way, both a single Assertions section or two sections (one for assertEqs, one for approx equals) is good with me. I don't feel strongly about this, but see the thumbs up from @ZeroEkkusu so will leave it up to you both 😛

Reply:
It ends up being 4 sections if you group overloaded functions together. This may be excessive and the argument for would be if we expect more overloads to come down the track - then this makes it blindingly obvious where to put them and whether they comply with existing style/format.

Will defer to @ZeroEkkusu to act as tie breaker and move this forward. For reference here is the commit where I group overloaded assertions together: 924456ad2a19fe4e504639988a856c58244079d2

Reply:
The idea I had was to use the big headers to divide the main parts of the contract:
"STD-CHEATS", "STD-ASSERTIONS", "STD-ERRORS", "STD-STORAGE".

I like the order as is, `assertEq` --> `assertApproxEqAbs`... It's logical IMO. Just put `assertFalse` first. I don't think we need any dividers rn; we can always add them later, as that'd be a non-breaking change.

Reply:
Okay, summary of changes:

- Collapse assertion code regions into one: 99a36d8f522b5b5db867d8633fb8ed887884f6de
- Re-order assertFalse: 3c15901e223e9c2b311d1c67c64c7d2e40e8615d
- Add STD-CHEATS code region: 113b11cc97dbe416db99b1e8dfdc8144a34126d2

Changes not done:

- Did not add STD-ERRORS or STD-STORAGE as these are standalone libraries, not regions within `contract Test`